### ISSUE TITLE
Admob banner won't show up on certain device only.

### ISSUE THREAD

**Issue by nxsm** [googleads/googleads-mobile-unity#230]
Hi, I made a script to show a banner ad on game over screen in android mobile 2d game. 
 The banner shows up in at least 4 different devices I have tested: 2 tablets and 2 low spec smartphones. 
 However when I am to test it on a new Huawei P8 Light, everything works just great however the ads wont show up.

Here what I have and tried so far: 

- My Test Device ID is 100% correct. Double checked it and tried 2 different apps from store to get it. ( Works for other devices just fine ).
- - Banner ID from admob site is correct. 
- Tried different AdSizes and also tried smartbanner but didn't work.
- Tried checking unity camera and layers for accidental overlay. All fine there. 
- Tried both  portrait and landscape for all devices. (All devices show banner, except this one I am having problem with ).
- Checked if this device got any adblocker. Nothing there.

My script is attached to the main camera which calls the request on Start() and hides on Update() till the game over panel shows up.

Here is my script, xml and adb logcat:

Script

` private BannerView bannerView;
  public GameObject GameOverPanel;
 
    void Start() {  AdRequest();  }

    private void AdRequest()
    {
	 
		bannerView = new BannerView("banner id is written here", AdSize.Banner, AdPosition.Bottom);

        AdRequest request = new AdRequest.Builder()
			.AddTestDevice("xxx") //Huawei P8
			.AddTestDevice("xxx")  // Tablet 
			.AddTestDevice("xxx") // Huawei IDEOS X5
			.AddTestDevice("xxx") // Tablet 2
            .SetGender(Gender.Unknown)
            .Build();

        bannerView.OnAdLoaded += HandleOnAdLoaded;
        bannerView.OnAdFailedToLoad += HandleOnAdFailedToLoad;
        bannerView.LoadAd(request);     
    }
 
    void Update()
    {
        if (GameOverPanel.activeSelf == true)
        {
            bannerView.Show();
        }
        else
        {
            bannerView.Hide();
        }
    }

	public void DestroyAd()
    {
        bannerView.Destroy();
    }

    public void HandleOnAdLoaded(object sender, EventArgs args)
	{
    }
    public void HandleOnAdFailedToLoad(object sender, EventArgs args)
    {
    }
 
`

manifest

```xml
<manifest xmlns:android="http://schemas.android.com/apk/res/android"
    package="com.google.unity"
    android:versionName="1.0"
    android:versionCode="1">
  <uses-sdk android:minSdkVersion="9"
      android:targetSdkVersion="19" />
  <!-- Google Mobile Ads Permissions -->
  <uses-permission android:name="android.permission.INTERNET"/>
  <uses-permission android:name="android.permission.ACCESS_NETWORK_STATE"/>
  <!-- Uncomment to add billing for in-app purchase ads -->
  <!--<uses-permission android:name="com.android.vending.BILLING"/> -->
  <application>
    <!-- Denote the referenced Google Play services version -->
    <meta-data android:name="com.google.android.gms.version"
        android:value="@integer/google_play_services_version" />
    <!-- Google Mobile Ads Activity -->
    <activity android:name="com.google.android.gms.ads.AdActivity"
        android:configChanges="keyboard|keyboardHidden|orientation|screenLayout|uiMode|screenSize|smallestScreenSize"
		   android:theme="@android:style/Theme.Translucent" />
 
  </application>
</manifest>
```

logcat

> I/ifdhandler smartcardapi(16237): IFDHICCPresence
I/ifdhandler proxy(16237): proxy_init - already connected to service manager
I/ifdhandler proxy(16237): proxy_isCardPresent - Lun 0
I/ifdhandler proxy(16237): proxy_readerlist_getReaderName - Lun 0
I/ifdhandler proxy(16237): proxy_readerlist_getReaderName Lun: 0, reader from list: SIM
I/ifdhandler ISmartcardAPI(16237):  isCardPresent
I/SmartcardSystemService( 3026): isCardPresent(SIM) returned: true
I/ifdhandler proxy(16237): readername used to check for card present: SIM,card present: true
I/ifdhandler proxy(16237): call to proxy_getLastError
I/ifdhandler ISmartcardAPI(16237):  getLastError
I/SmartcardSystemService( 3026): called: getLastError
I/ifdhandler proxy(16237): proxy_getLastError - no error
I/AdCheck (31247): APS: blockAdView: com.google.android.gms.ads.AdView{200cba06 G.E..... ......I. 0,0-0,0}
I/Ads     (31247): Scheduling ad refresh 60000 milliseconds from now.
I/AdCheck (31247): APS: blockAdView: com.google.android.gms.ads.AdView{200cba06 G.E..... ......I. 0,0-0,0}
I/Ads     (31247): Scheduling ad refresh 60000 milliseconds from now.
I/AdCheck (31247): APS: blockAdView: com.google.android.gms.ads.AdView{200cba06 G.E..... ......I. 0,0-0,0}
I/Ads     (31247): Scheduling ad refresh 60000 milliseconds from now.
I/AdCheck (31247): APS: blockAdView: com.google.android.gms.ads.AdView{200cba06 G.E..... ......I. 0,0-0,0}
I/Ads     (31247): Scheduling ad refresh 60000 milliseconds from now.
I/AdCheck (31247): APS: blockAdView: com.google.android.gms.ads.AdView{200cba06 G.E..... ......I. 0,0-0,0}
I/Ads     (31247): Scheduling ad refresh 60000 milliseconds from now.
I/AdCheck (31247): APS: blockAdView: com.google.android.gms.ads.AdView{200cba06 G.E..... ......I. 0,0-0,0}
I/Ads     (31247): Scheduling ad refresh 60000 milliseconds from now.
I/AdCheck (31247): APS: blockAdView: com.google.android.gms.ads.AdView{200cba06 G.E..... ......I. 0,0-0,0}
I/Ads     (31247): Scheduling ad refresh 60000 milliseconds from now.
I/AdCheck (31247): APS: blockAdView: com.google.android.gms.ads.AdView{200cba06 G.E..... ......I. 0,0-0,0}
I/Ads     (31247): Scheduling ad refresh 60000 milliseconds from now.
I/AdCheck (31247): APS: blockAdView: com.google.android.gms.ads.AdView{200cba06 G.E..... ......I. 0,0-0,0}
I/Ads     (31247): Scheduling ad refresh 60000 milliseconds from now.
I/AdCheck (31247): APS: blockAdView: com.google.android.gms.ads.AdView{200cba06 G.E..... ......I. 0,0-0,0}
I/Ads     (31247): Scheduling ad refresh 60000 milliseconds from now.
I/Ads     (31247): Ad is not visible. Not refreshing ad.
I/Ads     (31247): Scheduling ad refresh 60000 milliseconds from now.
I/AdCheck (31247): APS: blockAdView: com.google.android.gms.ads.AdView{200cba06 G.E..... ......I. 0,0-0,0}
I/Ads     (31247): Scheduling ad refresh 60000 milliseconds from now.
I/AdCheck (31247): APS: blockAdView: com.google.android.gms.ads.AdView{200cba06 G.E..... ......I. 0,0-0,0}
I/Ads     (31247): Scheduling ad refresh 60000 milliseconds from now.
I/AdCheck (31247): APS: blockAdView: com.google.android.gms.ads.AdView{200cba06 G.E..... ......I. 0,0-0,0}
I/Ads     (31247): Scheduling ad refresh 60000 milliseconds from now.
I/AdCheck (31247): APS: blockAdView: com.google.android.gms.ads.AdView{200cba06 G.E..... ......I. 0,0-0,0}
I/Ads     (31247): Scheduling ad refresh 60000 milliseconds from now.
I/Ads     (31247): Ad is not visible. Not refreshing ad.
I/Ads     (31247): Scheduling ad refresh 60000 milliseconds from now.
I/Ads     (31247): Ad is not visible. Not refreshing ad.
I/Ads     (31247): Scheduling ad refresh 60000 milliseconds from now.
I/Ads     (31247): Ad is not visible. Not refreshing ad.
I/Ads     (31247): Scheduling ad refresh 60000 milliseconds from now.
I/Ads     (31247): Ad is not visible. Not refreshing ad.
I/Ads     (31247): Scheduling ad refresh 60000 milliseconds from now.
I/Ads     (31247): Ad is not visible. Not refreshing ad.
I/Ads     (31247): Scheduling ad refresh 60000 milliseconds from now.
I/Ads     (31247): Ad is not visible. Not refreshing ad.
I/Ads     (31247): Scheduling ad refresh 60000 milliseconds from now.
I/Ads     (31247): Ad is not visible. Not refreshing ad.
I/Ads     (31247): Scheduling ad refresh 60000 milliseconds from now.
I/Ads     (31247): Ad is not visible. Not refreshing ad.
I/Ads     (31247): Scheduling ad refresh 60000 milliseconds from now.
I/Ads     (31247): Ad is not visible. Not refreshing ad.
I/Ads     (31247): Scheduling ad refresh 60000 milliseconds from now.
I/Ads     (31247): Ad is not visible. Not refreshing ad.
I/Ads     (31247): Scheduling ad refresh 60000 milliseconds from now.
W/cr_BindingManager(31247): Cannot call determinedVisibility() - never saw a connection for the pid: 31247
W/cr_ScreenOrientation(31247): Removing an inexistent observer!
I/Ads     (31247): Ad is not visible. Not refreshing ad.
I/Ads     (31247): Scheduling ad refresh 60000 milliseconds from now.
I/Ads     (31247): Ad is not visible. Not refreshing ad.
I/Ads     (31247): Scheduling ad refresh 60000 milliseconds from now.
I/Ads     (31247): Ad is not visible. Not refreshing ad.
I/Ads     (31247): Scheduling ad refresh 60000 milliseconds from now.
I/Ads     (31247): Ad is not visible. Not refreshing ad.
I/Ads     (31247): Scheduling ad refresh 60000 milliseconds from now.
I/Ads     (31247): Ad is not visible. Not refreshing ad.
I/Ads     (31247): Scheduling ad refresh 60000 milliseconds from now.
I/AdCheck (31247): APS: blockAdView: com.google.android.gms.ads.AdView{3585c08d G.E..... ......I. 0,0-0,0}
I/Ads     (31247): Starting ad request.
I/ifdhandler smartcardapi(16237): IFDHICCPresence
I/ifdhandler proxy(16237): proxy_init - already connected to service manager
I/ifdhandler proxy(16237): proxy_isCardPresent - Lun 0
I/ifdhandler proxy(16237): proxy_readerlist_getReaderName - Lun 0
I/ifdhandler proxy(16237): proxy_readerlist_getReaderName Lun: 0, reader from list: SIM
I/ifdhandler ISmartcardAPI(16237):  isCardPresent
I/SmartcardSystemService( 3026): isCardPresent(SIM) returned: true
I/ifdhandler proxy(16237): readername used to check for card present: SIM,card present: true
I/ifdhandler proxy(16237): call to proxy_getLastError
I/ifdhandler ISmartcardAPI(16237):  getLastError
I/SmartcardSystemService( 3026): called: getLastError
I/ifdhandler proxy(16237): proxy_getLastError - no error

Thanks in advance.

**Comment by rampara** [comment]
Are you able to reproduce the issue with the [sample application](https://github.com/googleads/googleads-mobile-unity/tree/master/samples/HelloWorld) in this repository? Using the linked sample application, I'm able to request and display AdMob banner ads on a Huawei P8 lite running Android 4.4.4 and EMUI 3.0.

**Comment by nxsm** [comment]
I don't have the phone right now but later tonight I'll try. Although it is strange as my app was tested in more than 5 different devices (Tablets and Smartphones) and all worked except Huawei P8 Lite. 
By the way thanks for the reply! After 10 days somebody finally answered :o

**Comment by rampara** [comment]
Are you sure the device doesn't have any sort of ad blocker installed on it? The logcat output under the tag "AdCheck" makes it seem like it might.

**Comment by rampara** [comment]
Closing issue due to non-response. If this issue persists, please re-open.

**Comment by armnotstrong** [comment]
Hi, I meet with the same issue with the same device mode (Huawei P8), I Have already reset the device to it's factory mode, so that I thinks it at least is not an AdBlocking app that do the mess. Maybe it's the EMUI is messing around with blocking stuff?

**Comment by rampara** [comment]
@armnotstrong this plugin does not log any messages under the `AdCheck` tag. If this behavior is not present on other devices for you, I would dig further into if there are any apps or features baked into the OS that performs ad blocking.

**Comment by Jiajie93** [comment]
I have the same problem. I got adCheck with huawei P8..

**Comment by rampara** [comment]
@Jiajie93 there may be some sort of ad blocker running on your particular device.

**Comment by Jiajie93** [comment]
@rampara I've noticed on logcat that when the log "**Initialized AdMob in container 11.3.02**" shows up, all my adRequest after this log will always fail. Showing "AdCheck: APS: blockAdView: com.google.android.gms.ads.AdView" as reason.
I'm using Huawei P8 as said before, I also asked to my friends to install my app (using their Huawei p8, Huawei p8 lite, and Huawei p9 lite) they all have the same problem..
Any idea? Digging on the internet I found another one having the same problem.. here is the link:
https://groups.google.com/forum/#!topic/google-admob-ads-sdk/t0nNOq3wTGg

**Comment by rampara** [comment]
@Jiajie93 the message you're seeing in logcat makes me believe your devices does have an ad blocker running on it.

**Comment by gammlstudio** [comment]
I have the same problem with Huawei P10, running Android 8.0.0, ads (banner, interstitial and video) aren't showing.  I have also tested on Huawei P6, Android 5.1.1 and on an older Prestigio phone, on both ads are displayed. Has this issue been solved or is is Huawei's devices problem from atleast P8 onwards?

**Comment by gammlstudio** [comment]
Issue is closed, but I will replay this anyway, because it seems I got my ads working again. Searched the web and found out that from the android 6.0 and onwards requesting permissions has changed, when it comes to user location. So in addition to Internet access permission, when using Admob, the location permission has to be requested as well. The issue was fixed for me when I turned on the location information from my phone. Turning it off prevents Admob ads from showing. Same thing happened with other downloaded apps from play store. So the problem is still there, when turning location info off, but for now I can go on testing my app. Maybe adding a permission request to android manifest will fix it.
